packages/vm/test/api/EIPs/eip-4200-static-relative.spec.ts

```diff
@@ -0,0 +1,222 @@
+  const len = code.length / 2
+  const lenHex = len.toString(16)
+  return magic + lenHex.padStart(4, '0') + '00' + code
+}
+
+tape('EIP 3670 tests', (t) => {
+  const common = new Common({
+    chain: Chain.Mainnet,
+    hardfork: Hardfork.London,
+    eips: [3540, 3670, 4200],
+  })
+
+  t.test('valid eip-4200 eof code', async (st) => {
+    const vm = await VM.create({ common })
+    const account = await vm.stateManager.getAccount(sender)
+    const balance = GWEI * BigInt(21000) * BigInt(10000000)
+    account.balance = balance
+    await vm.stateManager.putAccount(sender, account)
+    let nonce = 0
+
+    // Note: valid cases all have a format where the final opcode should be STOP
+    // There is only one STOP opcode in the code, which thus means that code
+    // arrived at the designated (wanted) stop opcode
+    // Note: this also verifies that these EOF codes are correct (otherwise code will not run)
+    const validCases = [
+      // RJUMP
+      // RJUMP, jump to JUMPDEST
+      [getEOFCode('5B5C' + getInt16Str(1) + '5B5B00'), 'RJUMP to JUMPDEST, +1'],
+      [getEOFCode('5B5C' + getInt16Str(0) + '5B5B00'), 'RJUMP to JUMPDEST, 0'],
+      // Note: need to jump over `JUMPDEST STOP`, otherwise an infinite loop is created
+      [
+        getEOFCode('5C' + getInt16Str(2) + '5B005C' + getInt16Str(-5) + '5B5B00'),
+        'RJUMP to JUMPDEST, -5',
+      ],
+      // RJUMP, jump to not-JUMPDEST (use `ADDRESS` (0x30) as dummy (valid) opcode)
+      [getEOFCode('305C' + getInt16Str(1) + '303000'), 'RJUMP to another opcode than JUMPDEST, +1'],
+      [getEOFCode('305C' + getInt16Str(0) + '303000'), 'RJUMP to another opcode than JUMPDEST, 0'],
+      // Note: need to jump over `JUMPDEST STOP`, otherwise an infinite loop is created
+      [
+        getEOFCode('5C' + getInt16Str(2) + '30005C' + getInt16Str(-5) + '5B5B00'),
+        'RJUMP to another opcode than JUMPDEST, -5',
+      ],
+      // RJUMPI
+      // RJUMPI, jump to JUMPDEST
+      [getEOFCode('5B60015D' + getInt16Str(1) + '5B5B00'), 'RJUMPI to JUMPDEST, +1'],
+      [getEOFCode('5B60015D' + getInt16Str(0) + '5B5B00'), 'RJUMPI to JUMPDEST, 0'],
+      // Note: need to jump over `JUMPDEST STOP`, otherwise an infinite loop is created
+      [
+        getEOFCode('5C' + getInt16Str(2) + '5B0060015D' + getInt16Str(-7) + '5B5B00'),
+        'RJUMPI to JUMPDEST, -7',
+      ],
+      // RJUMPI, jump to JUMPDEST
+      [getEOFCode('3060015D' + getInt16Str(1) + '303000'), 'RJUMPI to ADDRESS, +1'],
+      [getEOFCode('3060015D' + getInt16Str(0) + '303000'), 'RJUMPI to ADDRESS, 0'],
+      // Note: need to jump over `JUMPDEST STOP`, otherwise an infinite loop is created
+      [
+        getEOFCode('5C' + getInt16Str(2) + '300060015D' + getInt16Str(-7) + '303000'),
+        'RJUMPI to ADDRESS, -7',
+      ],
+      // RJUMPI: 0 is pushed on stack, so code will just continue after RJUMPI
+      [getEOFCode('60005D' + getInt16Str(1) + '006000FE'), 'RJUMPI, 0 is pushed on stack'],
```

Comment:
You are right regarding `FE` (INVALID), I am not sure why I added the PUSH 0 though because this makes no sense (I should update these test cases with notes)

Reply:
Actually, it does, it needs to put 0 on the stack (because RJUMPI pops one stack item). If I do not push0, then the stack is empty and thus stack underflows.